bundles/org.openhab.binding.mystrom/src/main/java/org/openhab/binding/mystrom/internal/AbstractMyStromHandler.java

```diff
@@ -91,30 +92,37 @@ public final void dispose() {
         super.dispose();
     }
 
-    private void updateProperties() throws MyStromException {
-        String json = sendHttpRequest(HttpMethod.GET, "/api/v1/info", null);
-        MyStromDeviceInfo deviceInfo = gson.fromJson(json, MyStromDeviceInfo.class);
-        if (deviceInfo == null) {
-            throw new MyStromException("Cannot retrieve device info from myStrom device " + getThing().getUID());
+    private void updateProperties() {
+        try {
+            String json = sendHttpRequest(HttpMethod.GET, "/api/v1/info", null);
+            MyStromDeviceInfo deviceInfo = gson.fromJson(json, MyStromDeviceInfo.class);
+            if (deviceInfo == null) {
+                throw new MyStromException("Cannot retrieve device info from myStrom device " + getThing().getUID());
+            }
+            this.mac = deviceInfo.mac;
+            Map<String, String> properties = editProperties();
+            properties.put(PROPERTY_MAC, deviceInfo.mac);
+            properties.put(PROPERTY_VERSION, deviceInfo.version);
+            properties.put(PROPERTY_TYPE, Long.toString(deviceInfo.type));
+            properties.put(PROPERTY_SSID, deviceInfo.ssid);
+            properties.put(PROPERTY_IP, deviceInfo.ip);
+            properties.put(PROPERTY_MASK, deviceInfo.mask);
+            properties.put(PROPERTY_GW, deviceInfo.gw);
+            properties.put(PROPERTY_DNS, deviceInfo.dns);
+            properties.put(PROPERTY_STATIC, Boolean.toString(deviceInfo.staticState));
+            properties.put(PROPERTY_CONNECTED, Boolean.toString(deviceInfo.connected));
+            Calendar calendar = Calendar.getInstance();
+            DateFormat formatter = DateFormat.getDateTimeInstance(DateFormat.SHORT, DateFormat.MEDIUM,
+                    Locale.getDefault());
+            properties.put(PROPERTY_LAST_REFRESH, formatter.format(calendar.getTime()));
+            updateProperties(properties);
+        } catch (Exception ex) {
```

Comment:
Please catch specific exceptions that can be thrown by your code.

Reply:
is this ok?
```
    } catch (JsonSyntaxException | MyStromException ex) {
        getLogger().debug("Updating properties failed: ", ex);
    }
```
Or should each exception be treated separately?

Reply:
It is ok